llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -2938,8 +3140,16 @@ bool AddressSanitizer::instrumentFunction(Function &F,
           if (CB->doesNotReturn())
             NoReturnCalls.push_back(CB);
         }
-        if (CallInst *CI = dyn_cast<CallInst>(&Inst))
-          maybeMarkSanitizerLibraryCallNoBuiltin(CI, TLI);
+        if (CallInst *CI = dyn_cast<CallInst>(&Inst)) {
+          if (TargetTriple.isSPIR() &&
+              CI->getCalledFunction()->getCallingConv() ==
```

Comment:
There are extensions (mostly work in progress) which add support for indirect calls to SYCL. In that case `CallInst::getCalledFunction()` will return `nullptr`  and trigger segfault here

Reply:
Done